### PR TITLE
Update MLOps stacks project layout

### PR DESCRIPTION
**Pull request by mingyu89**
Signed-off-by: Mingyu Li <mingyu.li@databricks.com>

stacks layout design doc https://docs.google.com/document/d/1BTFOzxiVzCJ2uKN0f9N3id85pL32Gof6c0cD1ao7fNo/edit#bookmark=id.gklrrat3qlf8

It's a big change. Let's focus on code correctness for now and fix/improve docs later.

Screen Shot 2023-02-21 at 5 06 37 PM

### Decisions
1. Create one more variable for the root folder. 
2. We'll just ask user what's the root folder name. Add to instructions that if user intends to use polyrepo, they should keep the root folder name same as project name.
3. Put terraform folder inside the project
4. Don't create a separate folder for projects, just add them to the root folder.

#### non-project related folders

- projects
    - .azure
        - devops-pipelines
        - …
    - .github
        - workflows
        - …
    - .mlops-setup-scripts
        - cicd
            - …
        - terraform
            - …
        - README.md
    - docs
        - images
            - ...
        - ...
    - README.md
    - pytest.ini
    - requirements.txt
    - test-requirements.txt
    - _params_testing_only.txt

#### project related folders

- projects
    - my_project1
        - terraform/bundle yaml.
        - utils.py
        - training
        - feature_engineering
        - validation
        - deployment
            - batch_inference
            - model_deployment
        - monitoring
        - data
            - sample.parquet
        - tests
            - training
            - feature_engineering
    - my_project2
    - my_project3
    - utils

---
# Test
All tests are passing
### run Train notebook
Screen Shot 2023-02-22 at 11 41 36 AM

### run batch inference
Screen Shot 2023-02-22 at 11 41 08 AM
Screen Shot 2023-02-22 at 11 43 35 AM
Screen Shot 2023-02-22 at 11 43 41 AM

### CODE REVIEW

**Comment by mingyu89** [comment]
Not ready for review

**Comment by mingyu89** [comment]
Ready for review

**Comment by mingyu89** [comment]
Ready for review again:) @vladimirk-db @zhe-db @arpitjasa-db

**Comment by mingyu89** [comment]
Tested running notebook and batch inference in databricks workspace.

**Review by arpitjasa-db — APPROVED** [review]
Awesome work @mingyu89 ! Would you mind linking us to the test project that you generated from this updated layout? It'll make it easier to confirm the structure is as expected

**Comment by mingyu89** [comment]
@arpitjasa-db I added you to my repo. It's the `refactor2` branch. https://github.com/mingyu89/test-repo1/tree/refactor2
Please note that the root folder name is not visible because it's also the git root folder.

File: cookiecutter.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
   "project_name": "my-mlops-project",
+  "root_dir__update_if_you_intend_to_use_monorepo": "{{cookiecutter.project_name}}",
```
[the remark this answers is not in the document]
Comment (vladimirk-db) [reply cookiecutter.json:3]: Can we document this in the readme? Add monorepo support blurb somewhere explicitly?

**Review by vladimirk-db — COMMENTED** [review]
Awesome - one comment on layout - I thought about this a little bit: can we actually remove the development folder and just pull up training (and feature_engineering, etc.) folders up a level?
This should help with the nesting and will make it clearer what types of pipelines are in there. "Development" sounds a bit vague, too.

File: cookiecutter.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mingyu89) [reply cookiecutter.json:3]: Sure. I have a ticket target at next sprint to address all doc issues. Most of them need to be updated.

**Comment by mingyu89** [comment]
@vladimirk-db I feel it better too to have one less layer for users. I made the change and tested by generating project and run the training and inference on databricks.

**Review by vladimirk-db — APPROVED** [review]
LGTM! Looks great, @mingyu89 .

**Pull request merged**